### PR TITLE
BD New Design - Search Components

### PR DESCRIPTION
**Pull request by shervElmi**
We will implement the search bar and advanced search designs with this PR.

#### Search Box with the region search:
https://www.loom.com/share/17fbc6c011534d579cb96daec525bcbc
#### Search Box with the zip code search:
https://www.loom.com/share/f68c77d9848940f38fef516a8c1837de

### Figma link:
https://www.figma.com/file/FelBhqZM5MVx7P3S4qikmr/Themes?node-id=1%3A2989&t=tROxcTXcpw5svkJp-4
https://www.figma.com/file/FelBhqZM5MVx7P3S4qikmr/Themes?node-id=1%3A2000&t=tROxcTXcpw5svkJp-4

### CODE REVIEW

**Comment by shervElmi** [comment]
@stephywells, I encountered a PHPStan error in `includes/admin/helpers/csv/class-csv-import.php`. Should we address this issue or configure PHPStan to ignore it? Please advise on the preferred approach.

**Comment by stephywells** [comment]
> @stephywells, I encountered a PHPStan error in `includes/admin/helpers/csv/class-csv-import.php`. Should we address this issue or configure PHPStan to ignore it? Please advise on the preferred approach.

I'm not seeing this file in the current Pr. Would you mind creating a new PR for that to fix the error? We can merge that into master to solve the problem.

**Comment by shervElmi** [comment]
> > @stephywells, I encountered a PHPStan error in `includes/admin/helpers/csv/class-csv-import.php`. Should we address this issue or configure PHPStan to ignore it? Please advise on the preferred approach.
> 
> I'm not seeing this file in the current Pr. Would you mind creating a new PR for that to fix the error? We can merge that into master to solve the problem.

I have now created a new PR specifically to address this issue.

PR URL: https://github.com/Strategy11/business-directory-plugin/pull/301

File: assets/css/less/front-end/elements/buttons.less [hunk 1]
```diff
@@ -19,18 +19,78 @@ body .wpbdp-excerpt {
 		display: inline-flex;
 		align-items: center;
 		justify-content: center;
+		border: 1px solid;
+		border-radius: 4px;
+		padding: 19px 40px;
```
Comment (stephywells) [inline assets/css/less/front-end/elements/buttons.less:24]: This is causing some issues from me in Astra. The button is taller than the inputs. 0 top and bottom padding solves it for me. 
Before:
Screenshot 2023-08-10 at 5 57 00 PM

After:
Screenshot 2023-08-10 at 5 57 16 PM

File: assets/css/less/front-end/elements/buttons.less [hunk 2]
```diff
@@ -19,18 +19,78 @@ body .wpbdp-excerpt {
 		display: inline-flex;
 		align-items: center;
 		justify-content: center;
+		border: 1px solid;
+		border-radius: 4px;
```
Comment (stephywells) [inline assets/css/less/front-end/elements/buttons.less:23]: Let's remove this line. I think we should let the theme handle the border radius so the buttons will match as closely as possible.

Without this line, here's what I see in Divi. The secondary button has the .et_pb_button class, but the main button doesn't have a class from the theme. Perhaps we need to apply the button class to the input submit too?
Screenshot 2023-08-10 at 6 06 42 PM

File: assets/css/less/front-end/components/submit-listing/plan-selection-section.less [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		}
+
+		.wpbdp-button {
+			width: 100%;
+		}
+	}
+
+	.wpbdp-msg {
+		display: flex;
+		display: none;
```
Comment (stephywells) [inline assets/css/less/front-end/components/submit-listing/plan-selection-section.less:67]: I'm not sure why this is hidden. Here's what I saw before:
Screenshot 2023-08-10 at 6 18 44 PM

But now nothing happens when I click, with no indication as to why it's not working.

**Review by stephywells — CHANGES_REQUESTED** [review]
I just started into this, but I think there will still be a lot of issues. Can you please send this PR through our regular review process? We'll need to compare before and after for every front-end facing page.
- select a plan
- submit a listing
- upload an image/file to a list
- see the confirmation page
- single listing
- list all
- advanced search
- (there might be more I'm not thinking of)

Thanks for the review and updates @mostafa-hisham !

**Comment by garretlaxton** [comment]
When I make a search using the advanced search in mobile view, the Find Listing button doesn't have it's width.
Before search:
image


After search:
image

**Comment by garretlaxton** [comment]
I'm still not seeing the width with the latest changes.

**Comment by garretlaxton** [comment]
Hmmm, that's weird. I did clear my cache and also tested with an incognito window and still don't see it.

**Review by garretlaxton — APPROVED** [review]
I uploaded this to a different site and managed to see the changes. Not sure why I wasn't seeing it locally, but this looks good!
image

**Review by stephywells — CHANGES_REQUESTED** [review]
Oops! I started a review and didn't submit it.

File: assets/css/less/submit-listing.less [hunk 4]
```diff
@@ -139,7 +139,7 @@
     .wpbdp-submit-listing-form-actions {
         margin: 10px 0;
         text-align: center;
-        
+
         button {
             margin: 15px;
```
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/css/less/submit-listing.less]: I already fixed this.

File: includes/controllers/pages/class-search.php [hunk 5]
```diff
@@ -53,11 +55,13 @@ public function dispatch() {
 			}
 		}
 
+		if ( $wpbdp_ajax_search_modal ) {
```
[the remark this answers is not in the document]
Comment (stephywells) [reply includes/controllers/pages/class-search.php]: This is fixed.

File: assets/css/less/front-end/components/submit-listing/rootline.less [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+				color: #fff;
+			}
+
+			/* Checked rootline name */
+			.wpbdp-rootline-section-name {
+				color: @color-text;
+			}
+
+			/* Current section */
+			&.wpbdp-submit-section-current {
```
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/css/less/front-end/components/submit-listing/rootline.less:84]: Fixed

File: assets/css/less/front-end/elements/forms.less [hunk 7]
```diff
@@ -23,140 +23,107 @@
...
+	textarea {
+		min-height: 120px;
 	}
 
-	.field-description {
-		color: @color-text--60;
-		font-size: 80%;
+	select,
+	.select2 .select2-selection {
+		background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='14' height='8' fill='none' viewBox='0 0 14 8'%3E%3Cpath stroke='%23000' stroke-linecap='round' stroke-linejoin='round' d='m1 1 6 6 6-6'/%3E%3C/svg%3E%0A") !important;
```
[the remark this answers is not in the document]
Comment (stephywells) [reply assets/css/less/front-end/elements/forms.less:44]: done.

File: assets/css/less/front-end/elements/buttons.less (hunk 2 above)
Comment (stephywells) [reply assets/css/less/front-end/elements/buttons.less]: Done

**Comment by stephywells** [comment]
@garretlaxton I would love it if you could look over this again. Here are the things we need to double check the appearance with multiple themes:
- All BD buttons
- Quick search
- Advanced search
- Select a plan
- Create a listing
- Upload an image in the listing
- Listing confirmation page

I think that should cover it.

**Comment by garretlaxton** [comment]
The renew listing and edit buttons on the manage listings page doesn't have any space between the 2. I see this on every theme I tested.
image

I think the X in the top right of the advanced search popup should be i little higher, it overlays the fields a little bit.
image

**Comment by stephywells** [comment]
Thanks for the finds! @garretlaxton can you tell me what theme (BD and WP) you were running for these screenshots?

**Comment by stephywells** [comment]
I saw the issue with the space between the edit/delete buttons and fixed that. I found a few more styling issues with buttons and spacing on the checkout page and delete listing page.

I've got a possible fix incoming for the modal X and I think I'm seeing the multiselect issue too. So no need on the theme names for right now.

**Review by stephywells — APPROVED** [review]

**Comment by shervElmi** [comment]
Yay! Thanks for your hard work on this, @stephywells! I'm so happy this PR is finally merged! You are awesome as always! 🚀 🪨

**Comment by stephywells** [comment]
Thank you for all your help with this too @shervElmi !

**Pull request merged**